### DISCUSSION TITLE
What is your ready to go `occ` command?

### DISCUSSION THREAD

**Discussion by ghost** [LADSoft/OrangeC#981] [General]
I found I could configure Geany to use Orange C. What I need is an optimal `occ` command. What is the `occ` command you use for production? Please give me both the commands for console application and window application. Thank you.

**Comment by LADSoft** [comment]
if you want to do a console program and not link against any dlls, just compile it like normal:

occ testprog.c

will produce and executable and

occ -c testprog.c

will produce an object file.

If you want to make a windowing application do this:

occ -Wg testprog.c

that will make an exe which uses WinMain and doesn't allow access to the console.

occ -Wg -c testprog.c

will just make the object file

for a dll do this:

occ -Wd testprog.c

and here you need DllMain
